src/main/java/org/openrewrite/java/testing/junit5/TemporaryFolderToTempDir.java

```diff
@@ -93,187 +100,181 @@ protected TemporaryFolderToTempDirVisitor getVisitor() {
+                    md = md.withParameters(parameters);
+                } else {
+                    md = md.withParameters(ListUtils.concatAll(md.getParameters(), parameters));
+                }
+                body = maybeAutoFormat(body, body.withStatements(statements), executionContext);
+                md = md.withBody(body);
             }
+            return md;
         }
+    }
 
+    private static class AddNewFolderMethod extends JavaIsoVisitor<ExecutionContext> {
+        private final J.MethodInvocation methodInvocation;
 
-        private static class AddNewFolderMethod extends JavaIsoVisitor<ExecutionContext> {
-            private final J.Identifier fieldName;
-            private final J.MethodInvocation methodInvocation;
+        public AddNewFolderMethod(J.MethodInvocation methodInvocation) {
+            this.methodInvocation = methodInvocation;
+        }
 
-            private AddNewFolderMethod(J.Identifier fieldName, J.MethodInvocation methodInvocation) {
-                this.fieldName = fieldName;
-                this.methodInvocation = methodInvocation;
-            }
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
 
-            @Override
-            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+            Stream<J.MethodDeclaration> methods = cd.getBody().getStatements().stream()
+                    .filter(J.MethodDeclaration.class::isInstance)
+                    .map(J.MethodDeclaration.class::cast);
+            boolean methodAlreadyExists = methods
+                    .anyMatch(m -> {
+                        List<Statement> params = m.getParameters();
+                        return m.getSimpleName().equals("newFolder")
+                                && params.size() == 2
+                                && params.get(0).hasClassType(FILE_TYPE)
+                                && params.get(1).hasClassType(STRING_TYPE);
+                    });
+            if (!methodAlreadyExists) {
+                cd = cd.withTemplate(template(
+                        "private static File newFolder(File root, String name) throws IOException {\n" +
+                                "    File result = new File(root, name);\n" +
+                                "    if(!result.mkdirs()) {\n" +
+                                "        throw new IOException(\"Couldn't create folders \" + root);\n" +
+                                "    }\n" +
+                                "    return result;\n" +
+                                "}"
+                ).imports("java.io.File", "java.io.IOException").javaParser(TEMPDIR_PARSER::get).build(), cd.getBody().getCoordinates().lastStatement());
+                maybeAddImport("java.io.File");
```

Comment:
After adding the method via the template, does the method have a MethodType (and is its declaring type correct)?

Reply:
I would assume that it does, can you elaborate on your concern?